### ISSUE TITLE
Projektstatus

### ISSUE THREAD

**Issue by mariomurrent-softwaresolutions** [krabina/Datencockpit#49]
Ist das Projekt eigentlich noch aktiv?

**Comment by Zabien** [comment]
Ja! Es geht in eine neue Runde, genauere Details gibt's ca. ab März!

**Comment by mariomurrent-softwaresolutions** [comment]
Uh sehr cool - Danke für die Info

**Comment by krabina** [comment]
Ich habe die öffentliche Insallation von www.datencockpit.at schon auf MediaWiki 1.35 angehoben. Funktioniert also schon fast alles (Wikieditor+ PageForms zicken noch etwas rum).

Mein Plan wäre eigentlich, gemeinsam mit einem technischen Partner eine Hosting-Lösung anzubieten, weil ich vermute, dass es hier schon noch Bedarf gibt. Viele Organisationen werden ja immer noch mit Excel-Sheets hantieren.

Ehrlich gesagt plane ich derzeit nicht, eine neue Release mit MW 1.35 hier anzubieten. Von 1.000 downloads der 1.27er-Version ist kein einziges Projekt entstanden.

D. h. bestehende Kunden können ihr Upgrade selbst machen, bei Bedarf kann ich ja gerne Unterstützung anbieten. Neue Kunden nehmen entweder eine gehostete Lösung oder fragen an, ob sie das on premise betreiben können. Was natürlich geht. Wichtig ist mir, dass es zu 100% auf Open-Source-Komponenten basiert.

Und immer noch würden wir auch einen Partner suchen, der sich mit Datenschutz auskennt. Also einen Datenschutzbeauftragten oder eine Beratungsfirma, die uns helfen könnte, wie man das inhaltlich noch besser machen kann bzw. die das ihren Kunden anbieten.

Kontaktaufnahme immer gerne über die Website.

**Comment by krabina** [comment]
Ich lassa das hier gerne offen als Thread. Ich weiß ja nicht mal, wer es überhaupt verwendet und bin an Meinungen zur Weiterentwicklung interessiert. Passt daher gut hierher.

**Comment by mariomurrent-softwaresolutions** [comment]
Danke für das ausführliche Feedback - eine Hosting Lösung die man dann quasi für Kunden anbieten könnte? Was genau sind denn die Anforderungen an einen Hosting Partner?

**Comment by krabina** [comment]
Erfahrung im (Semantic)MediaWiki hosting.

**Comment by devaxel** [comment]
Wie würde ich denn ein Update ausführen? Welche Schritte wären dazu nötig?

**Comment by krabina** [comment]
Im Grunde kann man der MediaWiki-Upgrade-Anleitung folgen:
https://www.mediawiki.org/wiki/Manual:Upgrading
https://www.semantic-mediawiki.org/wiki/Help:Upgrade/Upgrade_from_SMW_2.5_to_SMW_3.0

Also ein neues MW-Verzeichnis inkl. aller benötigten Extensions anlegen. Einzige Änderung: statt SemanticComments wird nun CommentStreams verwendet. Aus dem Original-Verzeichnis braucht man dann den "images"-Ornder sowie LocalSettings.php

Vermutlich wird es beim Ausführen von update.php ein paar Fehler geben. Hier hilft gegebenenfalls folgendes:

```
php maintenance/populateContentTables.php
php maintenance/cleanupUsersWithNoId.php --prefix=*
php extensions/SemanticMediaWiki/maintenance/updateEntityCountMap.php
php extensions/SemanticMediaWiki/maintenance/updateEntityCollation.php
```

Ich kann gerne (gegen ein kleines Entgelt) das komplette gezippte Verzeichnis meiner aktuellen Installation zur Verfügung stellen, dann erspart man sich das händische zusammenstellen der korrekten Versionen der Extensions. 

Änderungen, die dann noch in den Formularen und Vorlagen nötig geworden sind, kann man in den Änderungen an der öffentlichen Installation nachvollziehen (im Wesentlichen geht es da um Änderungen in Bootstrap (z. b. "card" statt "panel") und den Umstieg von Semantic Comments auf CommentStreams.

Zu beachten ist daher, dass User-Kommentare verloren gehen. Das müsste man den Usern dann gegebenenfalls kommunizieren.

**Comment by justarandomsysadmin** [comment]
Ein paar Punkte die ich nicht verstehe:
- Woher weißt du wie viele Projekte denn von den von dir genannten 1000 Downloads entstanden sind?
- Wieso stellst du nicht einfach die V2 allen zur Verfügung?
- Wieso muss ich mir jetzt selbst das Update zusammenbasteln?
- Wieso ist dir 100% Open Source wichtig obwohl du für ein Update Entgelt verlangst?

Nichts für ungut, aber es ist eine Sache neben dem Open Source Projekt eine SaaS-Version anzubieten für Menschen die es selbst nicht hosten können oder möchten. Updates aber jenen vorzuenthalten, die es bislang selbst gehostet haben und dafür nun ein Entgelt zu verlangen ist nichts anderes als es in ein Kostenpflichtige Software umzuwandeln. Da muss einfach das Kind beim Namen genannt werden, auch auf der Homepage anstatt einfach alte Irreführende Infos stehen zu lassen.

Alles in allem ist mir der Beigeschmack hier zu fad geworden und ich hoffe jemand macht einen Fork für die, die es weiterverwenden möchten. Ich für meinen Fall bin nie mit MediaWiki zurechtgekommen und nach kurzer Suche hab ich schon eine Alternative gefunden, die zudem noch aktiv und offen gepflegt wird.

Für alle anderen die sonst noch wechseln möchten: [Open Datenschutzcenter](https://open-datenschutzcenter.de/)

**Comment by krabina** [comment]
Hallo,
ich verstehe ich gut.Hier ein paar Antworten auf deine Fragen:
* Ich weiß nicht wieviele Projekte daraus entstanden sind. Ich vermute eben schon, dass 1.000 Downloads bedeuten, dass zumindest ein Teil davon das Datencockpit auch verwendet hat. Mit Projekte meinte ich irgendjemand von diesen Nutzer*innen, der mal Interesse an einem Zusatzfeature bekundet hätte oder an einem Upgrade, einer Installation oder irgendetwas, was zumindest ein Bisschen von eine Aufwand abgedeckt hat.
* Weil das ein Zusatzaufwand ist, den ich aktuell nicht unterbringe. Ich habe aktuell nicht mal noch die Hosting-Lösung umgesetzt. Habe nur www.datencockpit.at aktualisiert, damit ich hier nicht it alten Versionen am Start bin. In internen Installationen, die ev. nicht mal im Web erreichbar sind, spielt das ja auch weniger eine Rolle. 
* Weil ich noch nicht dazugekommen bin, etwas zu releasen. Ist einfach ein Zusatzaufwand. Die Netidee-Förderung ist lange vorbei und wie gesagt hat es leider nie Anfragen gegeben.
* Open Source bedeutet halt auch, dass sich entweder eine Community bilded, wo mehrere mitarbeiten und/oder es auch hin und wieder Leute gibt, die ein bisschen Budget aufstellen können. Meine Idee wäre, Budget durch ein Hosting anzubieten, aber wie gesagt hab ich nicht mal das geschafft, weil eben anere, bezahlte Jobs wichtiger sind als das.

Forken kann man natürlich, noch besser wären PRs. Oder auch einfach fragen, wie man das Projekt sonst unterstützen könnte.

Der Linktipp ist trotzdem interessant. Ich habe nie verstanden, warum es nicht mehr Open-Source-Lösungen für diesen Zweck gibt.

Ich poste hier sicher, sobald es was Neues gibt. Interessenten können sich weiterhin gerne melden. Ich hab das Projekt sicher noch nicht aufgegeben. :-)
